modules/imgproc/src/contours_common.hpp

```diff
@@ -159,23 +171,78 @@ class TreeIterator
     }
 
 private:
-    std::stack<int> levels;
     Tree<T>& tree;
+    std::stack<int> levels;
 };
 
 //==============================================================================
 
+template <typename T, size_t BLOCK_SIZE_ELEM, size_t STATIC_CAPACITY_BYTES>
+class ContourDataStorage
+{
+public:
+    typedef T data_storage_t;
+    typedef BlockStorage<data_storage_t, BLOCK_SIZE_ELEM, STATIC_CAPACITY_BYTES> storage_t;
+public:
+    ContourDataStorage(void) = delete;
```

Comment:
I'm wondering if these `=delete` and `=default` specifications are really needed in our case (also `TreeNode` and `Tree`). Perhaps we could omit them to reduce noise since these classes are internal and have limited use scope.

Reply:
`=delete` is a very good safeguard against bad usage